guides/source/association_basics.md

```diff
@@ -562,9 +562,9 @@ You are not free to use just any name for your associations. Because creating an
 
 Associations are extremely useful, but they are not magic. You are responsible for maintaining your database schema to match your associations. In practice, this means two things, depending on what sort of associations you are creating. For `belongs_to` associations you need to create foreign keys, and for `has_and_belongs_to_many` associations you need to create the appropriate join table.
 
-#### Creating Foreign Keys for `belongs_to` Associations
+#### Creating Foreign Keys and Indexes for `belongs_to` Associations
 
-When you declare a `belongs_to` association, you need to create foreign keys as appropriate. For example, consider this model:
+When you declare a `belongs_to` association, it's a good practice to add appropriate foreign keys to ensure referential data integrity and indexes to improve queries performance. For example, consider this model:
```

Comment:
> it's a good practice

You do need to create the key column, its not optional. Adding the constraint is a good practise.

Can you rephrase this part, and squash the commits.